src/Maths/Silk.NET.Maths/Matrix3x3.cs

```diff
@@ -0,0 +1,1157 @@
+        /// Row 2 of the matrix.
+        /// </summary>
+        [IgnoreDataMember]
+        public Vector3<T> Row2;
+        
+        /// <summary>
+        /// Row 3 of the matrix.
+        /// </summary>
+        [IgnoreDataMember]
+        public Vector3<T> Row3;
+
+        /// <summary>
+        /// Column 1 of the matrix.
+        /// </summary>
+        [IgnoreDataMember]
+        public Vector3<T> Column1 => new(Row1.X, Row2.X, Row3.X);
+        
+        
+        /// <summary>
+        /// Column 2 of the matrix.
+        /// </summary>
+        [IgnoreDataMember]
+        public Vector3<T> Column2 => new(Row1.Y, Row2.Y, Row3.Y);
+        
+        
+        /// <summary>
+        /// Column 3 of the matrix.
+        /// </summary>
+        [IgnoreDataMember]
+        public Vector3<T> Column3 => new(Row1.Z, Row2.Z, Row3.Z);
+
+        /// <summary>Value at row 1, column 1 of the matrix.</summary>
+        [DataMember]
+        public T M11
+        {
+            readonly get => Row1.X;
+            set => Row1.X = value;
+        }
+
+        /// <summary>Value at row 1, column 2 of the matrix.</summary>
+        [DataMember]
+        public T M12
+        {
+            readonly get => Row1.Y;
+            set => Row1.Y = value;
+        }
+
+        /// <summary>Value at row 1, column 3 of the matrix.</summary>
+        [DataMember]
+        public T M13
+        {
+            readonly get => Row1.Z;
+            set => Row1.Z = value;
+        }
+
+        /// <summary>Value at row 2, column 1 of the matrix.</summary>
+        [DataMember]
+        public T M21
+        {
+            readonly get => Row2.X;
+            set => Row2.X = value;
+        }
+
+        /// <summary>Value at row 2, column 2 of the matrix.</summary>
+        [DataMember]
+        public T M22
+        {
+            readonly get => Row2.Y;
+            set => Row2.Y = value;
+        }
+
+        /// <summary>Value at row 2, column 3 of the matrix.</summary>
+        [DataMember]
+        public T M23
+        {
+            readonly get => Row2.Z;
+            set => Row2.Z = value;
+        }
+
+        /// <summary>Value at row 3, column 1 of the matrix.</summary>
+        [DataMember]
+        public T M31
+        {
+            readonly get => Row3.X;
+            set => Row3.X = value;
+        }
+
+        /// <summary>Value at row 3, column 2 of the matrix.</summary>
+        [DataMember]
+        public T M32
+        {
+            readonly get => Row3.Y;
+            set => Row3.Y = value;
+        }
+
+        /// <summary>Value at row 3, column 3 of the matrix.</summary>
+        [DataMember]
+        public T M33
+        {
+            readonly get => Row3.Z;
+            set => Row3.Z = value;
+        }
+        
+        /// <summary>
+        /// Indexer for the rows of this matrix.
+        /// </summary>
+        /// <param name="i">The row to select. Zero based.</param>
+        public unsafe Vector3<T> this[int i]
```

Comment:
It's not clear whether this will return a row or a column from the parameter name.
```suggestion
        public unsafe Vector3<T> this[int x]
```

Reply:
Well these are row major, but I can still change it anyways.